packages/clerk-js/src/ui/utils/createCustomMenuItems.tsx

```diff
@@ -0,0 +1,226 @@
+import type { CustomMenuItem, LoadedClerk } from '@clerk/types';
+
+import { USER_BUTTON_ITEM_ID } from '../constants';
+import type { LocalizationKey } from '../customizables';
+import { CogFilled, SignOut } from '../icons';
+import { localizationKeys } from '../localization';
+import { ExternalElementMounter } from './ExternalElementMounter';
+import { isDevelopmentSDK } from './runtimeEnvironment';
+import { sanitizeCustomLinkURL } from './sanitizeCustomLinkURL';
+
+export type DefaultItemIds = 'manageAccount' | 'addAccount' | 'signOut' | 'signOutAll';
+
+export type MenuItem = {
+  id: string;
+  name: LocalizationKey | string;
+  icon: React.ComponentType;
+  onClick?: () => void;
+  open?: string;
+} & ({ external: true; path: string } | { id: string; external?: false; path?: string });
```

Comment:
Seem like we are already declaring `id` in line 14
```suggestion
} & ({ external: true; path: string } | { external?: false; path?: string });
```

Reply:
Can't we just infer that something is external from simply path being defined or undefined ?
```suggestion
} & ({  path?: string });
```

Reply:
Correct @panteliselef , I fixed that